resources/js/views/settings/users/View.vue

```diff
@@ -0,0 +1,510 @@
+        this.hasNextPage = page < response.data.meta.last_page;
+        return this.modelLoadPromise;
+      }).then(() => {
+        this.roles.forEach(role => {
+          role.$isDisabled = !!this.model.roles.find(selectedRole => selectedRole.id === role.id && selectedRole.automatic);
+        });
+      }).catch(error => {
+        Base.error(error, this.$root, error.message);
+      }).finally(() => {
+        this.rolesLoading = false;
+      });
+    },
+
+    /**
+     * Saves the changes of the user to the database by making a api call.
+     *
+     * @param evt
+     */
+    saveUser (evt) {
+      if (evt) {
+        evt.preventDefault();
+      }
+
+      this.isBusy = true;
+
+      const config = {
+        method: this.config.id === 'new' ? 'post' : 'put',
+        data: {
+          firstname: this.model.firstname,
+          lastname: this.model.lastname,
+          username: this.model.username,
+          email: this.model.email,
+          password: this.model.password,
+          password_confirmation: this.model.password_confirmation,
+          user_locale: this.model.user_locale,
+          roles: this.model.roles.map(role => role.id),
+          updated_at: this.model.updated_at
+        }
+      };
+
+      Base.call(this.config.id === 'new' ? 'users' : `users/${this.config.id}`, config).then(response => {
+        const localeChanged = this.$store.state.session.currentLocale !== config.data.user_locale;
+
+        // if the updated user is the current user, then renew also the currentUser by calling getCurrentUser of the store
+        if (PermissionService.currentUser && this.config.id === PermissionService.currentUser.id) {
+          return this.$store.dispatch('session/getCurrentUser').then(() => {
+            if (localeChanged) {
+              return loadLanguageAsync(config.data.user_locale).then(() => {
+                this.$store.commit('session/setCurrentLocale', config.data.user_locale);
+                return response;
+              });
+            }
+
+            return Promise.resolve(response);
+          });
+        }
+
+        return Promise.resolve(response);
+      }).then(response => {
+        // don't change page on save on profile page
+        if (this.config.type !== 'profile') {
+          this.$router.push({ name: 'settings.users' });
+        } else {
+          this.model = response.data.data;
+          this.roles.forEach(role => {
+            role.$isDisabled = !!this.model.roles.find(selectedRole => selectedRole.id === role.id && selectedRole.automatic);
+          });
+          this.model.roles.forEach(role => {
+            role.$isDisabled = role.automatic;
+          });
+        }
+      }).catch(error => {
+        // If the user wasn't found and it is the current user log him out!
+        if (PermissionService.currentUser && this.config.id === PermissionService.currentUser.id && error.response && error.response.status === env.HTTP_NOT_FOUND) {
```

Comment:
what if user isn't found but is not the current user?

Reply:
In this case the user will get an 404 error shown by the global error handler. This is not as critical as if the currentUser can't be found.